pkg/api/pinning_bzz.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/traversal"
+	"github.com/gorilla/mux"
+)
+
+// pinBzzUploaded is used to pin an already uploaded content.
+func (s *server) pinBzzUploaded(w http.ResponseWriter, r *http.Request) {
+	addr, err := swarm.ParseHexAddress(mux.Vars(r)["address"])
+	if err != nil {
+		s.Logger.Debugf("pin bzz: parse address: %v", err)
+		s.Logger.Error("pin bzz: parse address")
+		jsonhttp.BadRequest(w, "bad address")
+		return
+	}
+
+	has, err := s.Storer.Has(r.Context(), addr)
+	if err != nil {
+		s.Logger.Debugf("pin bzz: localstore has: %v", err)
+		s.Logger.Error("pin bzz: store")
+		jsonhttp.InternalServerError(w, err)
+		return
+	}
+
+	if !has {
+		jsonhttp.NotFound(w, nil)
+		return
+	}
+
+	ctx := r.Context()
+
+	chunkAddressFn := func(address swarm.Address) (stop bool) {
```

Comment:
this function should be the same across all API calls maybe and taken from the `pin` package